dora/core/client/fs/src/main/java/alluxio/client/file/cache/LocalCacheFileInStream.java

```diff
@@ -204,6 +205,9 @@ private int localCachedRead(ReadTargetBuffer bytesBuffer, int length,
             stopwatch.elapsed(TimeUnit.NANOSECONDS));
       }
       return bytesRead;
+    } else {
+      //In case any error in cache manager, revert the offset change in the bytesBuffer
+      bytesBuffer.offset(offsetInBuffer);
```

Comment:
will `alluxio.client.file.cache.LocalCacheManager#getAndLoad` share the same issue like here?
as a result, maybe we shall store and reset the buffer offset passed in inside `alluxio.client.file.cache.LocalCacheManager#get(alluxio.client.file.cache.PageId, int, int, alluxio.file.ReadTargetBuffer, alluxio.client.file.CacheContext)` to fully resolve this issue?

Reply:
good call, I shiftedthe logic to cache manager. thanks!